scrapy/core/downloader/handlers/http11.py

```diff
@@ -41,7 +42,24 @@ def download_request(self, request, spider):
         return agent.download_request(request)
 
     def close(self):
-        return self._pool.closeCachedConnections()
+        d = self._pool.closeCachedConnections()
+        # closeCachedConnections will hang on network or server issues, so
+        # we'll manually timeout the deferred.
+        #
+        # Twisted issue addressing this problem can be found here:
+        # https://twistedmatrix.com/trac/ticket/7738.
+        #
+        # closeCachedConnections doesn't handle external errbacks, so we'll
+        # issue a callback after `_disconnect_timeout` seconds.
+        delayed_call = reactor.callLater(self._disconnect_timeout, d.callback, [])
```

Comment:
my understanding is that while this will "unblock" the deferred, it will left the connection opened...
so you should also do something like self.transport.abortConnection()